### ISSUE TITLE
render/vulkan: add support for VK_EXT_external_memory_host

### ISSUE THREAD

**Issue by emersion** [swaywm/wlroots#3264]
This would allow for non-blocking shm texture uploads.

---

wlroots has migrated to gitlab.freedesktop.org. This issue has been moved to:

https://gitlab.freedesktop.org/wlroots/wlroots/-/issues/3264

**Comment by unrelentingtech** [comment]
@jbeich yes, radv would be affected of course. We really should implement mmu notifiers already…

**Comment by ascent12** [comment]
>This would allow for non-blocking shm texture uploads.

Isn't uploading stuff to the GPU just a normal command submission, that isn't blocking anyway? It just depends how you wait on the fence or whatever.
All I think this extension would allow is potentially skipping a copy of the client's contents into a staging buffer.

**Comment by emersion** [comment]
This `memcpy` is blocking: https://github.com/swaywm/wlroots/blob/cc2ebd9fc0bd3e304e281b73e638824bc1c7332b/render/vulkan/texture.c#L97

The ext allows Vulkan to asynchronously perform that `memcpy`, but I think the staging buffer will stay since we need HOST_VISIBLE | HOST_COHERENT?
